### ISSUE TITLE
Bind service activation with a keyboard shortcut

### ISSUE THREAD

**Issue by TiBeN** [systemd/systemd#7452]
### Submission type

 - Request for enhancement (RFE)

### systemd version the issue has been seen with

> 234.9

### Used distribution

> Fedora 27

In systemd there are many ways to activate a service but not with keyboard shortcut, or i have not found how to (in this case my apologies).

The special target `ctrl-alt-del.target` relates to this. It could be great if we would be able to define our own keyboard shortcut target, for example: `ctrl-shift-f3.target`. This is of course just a design idea of a end user point of view.

**Comment by poettering** [comment]
Hmm, sorry, but keypress handling is stuff for user facing components, such as gnome, wayland, x11, ... We only handle the two magic keypresses the kernel treats specially: Ctrl-Alt-Del, and Ctrl-Alt-ArrowUp, and they are intended to be the last resort if everything fails, only.

Or to say this differently: either convince the kernel folks that they make more key combinations like that special (but that's unlikely to work), or ask the UI people to add infrastructure for this. But systemd is the wrong place...

Sorry.

**Comment by TiBeN** [comment]
OK, thank you for your quick reply.

My specific use case was to be able to use a keyboard shortcut whatever the running software stack (bash, x11, fullscreen app etc.). I think it should be done with a simple daemonized piece of code that listens to keyboard events.

Thanks

**Comment by maelstrom256** [comment]
I faced this feature as viable, suddenly.
My setup is: ubuntu 18.04, 2 GPU, 2 Windows guests. Until one gpu gone, it was simple to have desktop at one GPU and two shortcuts for launch two VMs, share remaining GPU and USB-controller.
Now, with one GPU remains, I set up systemd to multi-user target instead graphical target, and need another ssh-capable device to start VMs via console or virt-manager. Or to login to console just to start VM.
Ridiculous, isn't? Keyboard and mouse plugged in shared USB-controller, so using some key-shortcut to launch linux desktop, VM1 or VM2 will be nice to have.
And much better, if it can be systemd-unit solution, rather than script crunch.
